src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -101,6 +102,9 @@ var DOMRenderer = ReactFiberReconciler({
     props : Props,
     rootContainerInstance : Container,
   ) : void {
+    if (__DEV__) {
+      validateProperties(domElement, type, null, props);
```

Comment:
Why don't we just set `ReactDebugCurrentFiber` during the commit phase too so that it's easy for any renderer to get to this debug information during whatever validation they want?